### ISSUE TITLE
[BUG] Rancher throws panic when viewing API paths with insufficient permissions in DEBUG mode

### ISSUE THREAD

**Issue by HarrisonWAffel** [rancher/rancher#40296]
**Rancher Server Setup**
- Rancher version: 2.7-head
- Installation option (Docker install/Helm Chart): IDE
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): n/a
- Proxy/Cert Details: n/a

**Information about the Cluster**
- Kubernetes version: 1.25
- Cluster Type (Local/Downstream): Local
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): n/a

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Standard user, no roles or privileges attached
  - If custom, define the set of permissions:

**Describe the bug**
When viewing particular API paths as a standard user a panic log can be observed in the rancher server logs. 

**To Reproduce**
+ Start Rancher with the `--debug` flag
+ Create a standard user with no roles or privileges 
+ Login as that standard user 
+ navigate to `v1/management.cattle.io.globalroles/podsecurityadmissionconfigurationtemplates-manage`

So far I've been able to reproduce this the `management.cattle.io.globalroles` API, but there may be other paths effected too. 

**Result**

look at the rancher server logs, and observe a panic message

**Expected Result**
The user is denied access, but a panic is not seen in the rancher logs when `--debug` mode is enabled

**Additional context**

<details closed>
<summary>
Panic Logs
</summary>

```
2023/01/24 18:37:15 [DEBUG] 2023/01/24 18:37:15 http: panic serving [::1]:62441: runtime error: invalid memory address or nil pointer dereference
2023/01/24 18:37:15 [DEBUG] goroutine 21164 [running]:
2023/01/24 18:37:15 [DEBUG] net/http.(*conn).serve.func1()
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:1850 +0x118
2023/01/24 18:37:15 [DEBUG] panic({0x1085c5d80, 0x10b9984a0})
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/runtime/panic.go:890 +0x26c
2023/01/24 18:37:15 [DEBUG] github.com/rancher/apiserver/pkg/server.(*Server).handle(0x140017639b0, 0x140032d7c20, 0x109095418)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20230120214941-e88c32739dc7/pkg/server/server.go:154 +0x64c
2023/01/24 18:37:15 [DEBUG] github.com/rancher/apiserver/pkg/server.(*Server).Handle(0x140017639b0, 0x140032d7c20)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20230120214941-e88c32739dc7/pkg/server/server.go:101 +0x38
2023/01/24 18:37:15 [DEBUG] github.com/rancher/steve/pkg/server/handler.(*apiServer).apiHandler.func1({0x1090ef3d0, 0x14001182000}, 0x14008b03100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/steve@v0.0.0-20230120232010-53fbb87f5968/pkg/server/handler/apiserver.go:97 +0xc4
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140018c2bd0, {0x1090ef3d0, 0x14001182000}, 0x14008b03100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1({0x1090ef3d0, 0x14001182000}, 0x14008b03100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/steve@v0.0.0-20230120232010-53fbb87f5968/pkg/auth/filter.go:172 +0x4ac
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x14001763b30, {0x1090ef3d0, 0x14001182000}, 0x14008b03000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x1090ef3d0, 0x14001182000}, 0x14008b03000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20230120214941-e88c32739dc7/pkg/urlbuilder/redirect.go:14 +0x88
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x14003b512d8, {0x1090ef3d0, 0x14001182000}, 0x14008b03000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140010f0e40, {0x1090ef3d0, 0x14001182000}, 0x14008b03000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/auth/requests.authedFilter.ServeHTTP({{0x1090b0ea0, 0x140014d88f0}, {0x10742156e, 0x4}, {0x1400092f810, 0x1, 0x1}}, {0x1090ef3d0, 0x14001182000}, 0x14008b02e00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/auth/requests/filter.go:105 +0x1c8
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140010f1140, {0x1090ef3d0, 0x14001182000}, 0x14008b02e00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140010f1200, {0x1090ef3d0, 0x14001182000}, 0x14008b02c00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140002306c0, {0x1090ef3d0, 0x14001182000}, 0x14008b02a00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x14000231440, {0x1090ef3d0, 0x14001182000}, 0x14008b02600)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x14000231a40, {0x1090ef3d0, 0x14001182000}, 0x14008b02200)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x14000231980, {0x1090ef3d0, 0x14001182000}, 0x14008b02000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x14000231800, {0x1090ef3d0, 0x14001182000}, 0x14001cf3e00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/multiclustermanager.(*DeferredServer).Middleware.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3b00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/multiclustermanager/deferred.go:113 +0x88
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x1400030a6c0, {0x1090ef3d0, 0x14001182000}, 0x14001cf3b00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140010f1080, {0x1090ef3d0, 0x14001182000}, 0x14001cf3b00)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140067163c0, {0x1090ef3d0, 0x14001182000}, 0x14001cf3600)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/api/steve/aggregation.(*aggregationHandler).Middleware.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3400)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/api/steve/aggregation/aggregation.go:52 +0x70
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x1400030a700, {0x1090ef3d0, 0x14001182000}, 0x14001cf3400)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/gorilla/mux.(*Router).ServeHTTP(0x140010f0fc0, {0x1090ef3d0, 0x14001182000}, 0x14001cf3400)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x258
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/api/steve/proxy.NewProxyMiddleware.func1.1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/api/steve/proxy/proxy.go:76 +0x88
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x1400030a720, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/api/steve/proxy.RewriteLocalCluster.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/api/steve/proxy/proxy.go:43 +0x17c
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140018c3170, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG] Triggering auth refresh on u-wfzcj
2023/01/24 18:37:15 [DEBUG] Skipping refresh for u-wfzcj due to max-age
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP({{0x1090bdf60, 0x140018c3170}}, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/websocket/handler.go:35 +0xf4
2023/01/24 18:37:15 [DEBUG] github.com/rancher/apiserver/pkg/middleware.NoCache.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20230120214941-e88c32739dc7/pkg/middleware/cache.go:33 +0x90
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140018c3188, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/apiserver/pkg/middleware.ContentTypeOptions.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20230120214941-e88c32739dc7/pkg/middleware/cache.go:47 +0x90
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140018c31a0, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/auth.SetXAPICattleAuthHeader.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/auth/server.go:180 +0x164
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140018c31b8, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/rancher/pkg/auth/audit.auditHandler.ServeHTTP({{0x1090bdf60, 0x140018c31b8}, 0x0, 0x14001775680, 0x14004e20240, 0x140043a5da0}, {0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/fork/rancher/pkg/auth/audit/filter.go:65 +0x57c
2023/01/24 18:37:15 [DEBUG] github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1({0x1090ef3d0, 0x14001182000}, 0x14001cf3100)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/steve@v0.0.0-20230120232010-53fbb87f5968/pkg/auth/filter.go:172 +0x4ac
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x14004e202a0, {0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/dynamiclistener/server.wrapHandler.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/dynamiclistener@v0.3.5/server/server.go:193 +0x84
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x14004e44210, {0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] github.com/rancher/dynamiclistener.HTTPRedirect.func1({0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/go/pkg/mod/github.com/rancher/dynamiclistener@v0.3.5/redirect.go:20 +0x2d8
2023/01/24 18:37:15 [DEBUG] net/http.HandlerFunc.ServeHTTP(0x140039981f8, {0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2109 +0x44
2023/01/24 18:37:15 [DEBUG] net/http.serverHandler.ServeHTTP({0x140032d6690}, {0x1090ef3d0, 0x14001182000}, 0x14001cf3000)
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:2947 +0x41c
2023/01/24 18:37:15 [DEBUG] net/http.(*conn).serve(0x140097c80a0, {0x1090f2b70, 0x14007e0bd00})
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:1991 +0x14e0
2023/01/24 18:37:15 [DEBUG] created by net/http.(*Server).Serve
2023/01/24 18:37:15 [DEBUG]     /Users/haffel/sdk/go1.19.1/src/net/http/server.go:3102 +0x7e4
```

</details>

**Comment by cbron** [comment]
Added to q3 but there is no priority

**Comment by samjustus** [comment]
@cbron  why not q2?

**Comment by cbron** [comment]
We can do q2 if we have time.

**Comment by sowmyav27** [comment]
@JonCrowther Could you include the QA Test template for this issue?

**Comment by Priyashetty17** [comment]
I was able to recreate this issue on v2.7.3 for several API paths that the user did not have access to. I have validated the fix on v2.7-5334be362e291576c8f32b4432762399729832e6-head. 

## ✅ PASSED ##

| Component                        | Version         |
| -------------------------------- | ------------------------ |
| Rancher version                  | v2.7-5334be362e291576c8f32b4432762399729832e6-head                         |
| Dashboard              | master db8d42c1b |


**Test Steps:**
1. Configure Rancher to capture DEBUG logs.
2. Log in as an admin.
3. Access a non-existent API path and verify that Rancher doesn't panic. 
4. Create a standard user and log in using that user's credentials.
5. Access each of the following API paths, for which the standard user doesn't have permission, and verify that Rancher doesn't panic.
Note: In v2.7.3, prior to the fix, accessing these API paths caused Rancher to panic.
    - v1/management.cattle.io.globalroles/podsecurityadmissionconfigurationtemplates-manage
    - v1/management.cattle.io.apiservices
    - v1/management.cattle.io.authconfigs
    - v1/management.cattle.io.cisbenchmarkversions
    - v1/management.cattle.io.cisconfigs
    - v1/management.cattle.io.clusteralertgroups
    - v1/management.cattle.io.clusteralertrules
    - v1/management.cattle.io.clusteralerts
    - v1/management.cattle.io.clustercatalogs
    - v1/management.cattle.io.clusterloggings
    - v1/management.cattle.io.clustermonitorgraphs
    - v1/management.cattle.io.clusterregistrationtokens
    - v1/management.cattle.io.clusterroletemplatebindings
    - v1/management.cattle.io.clusterscans
    - v1/management.cattle.io.clustertemplates
    - v1/management.cattle.io.composeconfigs
    - v1/management.cattle.io.dynamicschemas
    - v1/management.cattle.io.etcdbackups
    - v1/management.cattle.io.globalrolebindings
    - v1/management.cattle.io.globalroles
    - v1/management.cattle.io.groupmembers
    - v1/management.cattle.io.groups
    - v1/management.cattle.io.managedcharts
    - v1/management.cattle.io.monitormetrics
    - v1/management.cattle.io.multiclusterapprevisions
    - v1/management.cattle.io.nodepools
    - v1/management.cattle.io.nodes
    - v1/management.cattle.io.notifiers
    - v1/management.cattle.io.podsecuritypolicytemplateprojectbindings
    - v1/management.cattle.io.projectalertgroups
    - v1/management.cattle.io.projectalertrules
    - v1/management.cattle.io.projectalerts
    - v1/management.cattle.io.projectcatalogs
    - v1/management.cattle.io.projectloggings
    - v1/management.cattle.io.projectmonitorgraphs
    - v1/management.cattle.io.projectnetworkpolicies
    - v1/management.cattle.io.projectroletemplatebindings
    - v1/management.cattle.io.samltokens
    - v1/management.cattle.io.userattributes
